### ISSUE TITLE
Bug: Tax 2 not calculated on suppliers orders invoices

### ISSUE THREAD

**Issue by mrsarge** [Dolibarr/dolibarr#6471]
# Bug
In suppliers orders, second tax is applied in the order and draft invoice, but not on final invoice. everything is fine on customers orders.

## Environment
- **Version**: [4.0.4] - [8.0.4]
- **OS**: [Ubuntu 16.04.1]
- **Web server**: [Apache 2.4.18]
- **PHP**: [PHP 5.6.31]
- **Database**: [MYSQL 5.7.24]

## Expected and actual behavior
Tax 2 amount should be shown on invoice and total due should include it.  

## Steps to reproduce the behavior
Create a new supplier third party with two tax.  
Create new supplier order for this third party.  
Add product or service with sales tax.  
Validate and approve order. Tax, Tax 2 and Total should be ok.   
Create invoice.  
Tax 2 line is not shown, but Total include it.  
Create draft.  
Tax 2 is 0.00 $ and Total doesn't include it.  
The invoice PDF also has a 0.00 Tax 2 line and Total doesn't include it either.

## Screenshots  
**Order page and PDF:**
![order](https://user-images.githubusercontent.com/6961825/30011418-2839dbac-9105-11e7-9733-0f9deab29d33.png)
![30011420-2a502324-9105-11e7-9e91-af815832bd52](https://user-images.githubusercontent.com/6961825/31409808-1c08f1f4-addb-11e7-8ffa-661f881c1928.png)

**Invoice page and PDF:**
![invoice](https://user-images.githubusercontent.com/6961825/30011429-377e8900-9105-11e7-9404-838cf41dded7.png)
![invoice_pdf](https://user-images.githubusercontent.com/6961825/30011430-39270976-9105-11e7-890b-0a5ef350b2b8.png)

**Comment by mrsarge** [comment]
Just upgraded to 5.0.2 and issue is still there.

**Comment by technoartista** [comment]
How to enable a second and 3rd tax for the already added third party . I can see it allows me to edit for new 3rd party but not allowing me to edit for existing 3rd party .

**Comment by mrsarge** [comment]
Issue still present in 6.0.0.  Maybe similar to issue #7298

**Comment by mrsarge** [comment]
Issue still present in 6.0.2.

**Comment by mrsarge** [comment]
Partially fixed in Dolibarr 6.0.3 (same for 6.0.4).

The total on the invoice page and pdf (validated or not) is still wrong (second tax missing). 

BUT, if you modify or add/remove a product from the invoice page, the amount is now correctly calculated and the PDF is ok.

See screenshot of the invoice page, before and after I added a product.

![chrome_2017-12-13_17-44-38](https://user-images.githubusercontent.com/6961825/33966539-ef5a7918-e02d-11e7-866a-1205c9a6d16f.png)

![chrome_2017-12-13_17-45-15](https://user-images.githubusercontent.com/6961825/33966542-f3648c92-e02d-11e7-88fc-53654f3d0b4e.png)

**Comment by mrsarge** [comment]
Tested all of the following versions and they still have the issue:

6.0.5, 6.0.6, 6.0.7, 6.0.8
7.0.0, 7.0.1, 7.0.3, 7.0.4
8.0.0, 8.0.1, 8.0.2, 8.0.3, 8.0.4

**Comment by nanoxp** [comment]
the following versions  they still have the issue: 9.0.0, 9.0.1,9.0.2 y 9.0.3 
Mr. Developers of Dolibarr the second tax is used in many countries and carries many versions with the problem, this limits the full use of the Dolibarr, when do you plan to fix it?

**Comment by nanoxp** [comment]
They must activate in the supplier's fiche that If you handle the second tax so that it can already appear and calculate it in the purchase order or in the invoice supplier el taxes 2
![imagen](https://user-images.githubusercontent.com/24461406/59454225-3cdb4980-8dd7-11e9-9919-5ae511c7fdb7.png)

**Comment by mrsarge** [comment]
The supplier's I am using for my tests, as stated in my original post, has the second tax enabled.

![chrome_2019-06-13_18-18-39](https://user-images.githubusercontent.com/6961825/59471038-cd318280-8e07-11e9-886f-d141c8b53411.png)

**Comment by simnandez** [comment]
I have tested it in Dolibarr 9, 10 and 11 and works fine

**Comment by mrsarge** [comment]
I tested 8.05 and 8.0.6, still have the issue.
I upgraded to 9 and tested 9.0.0, 9.0.1, 9.0.2 and 9.0.3, still have the issue.

Finally tested 9.0.4 and it fixed the issue!

It took 2 years and 5 major version, but this bug is now fixed.
Thanks to whoever fixed it!
I don't use it anymore due to this bug, but I kept my instance so I could continue to update this issue.

(10.0.0 to 10.0.6 are also fixed. Didn't try the 11 branch.)
